sdks/python/apache_beam/transforms/external.py

```diff
@@ -655,22 +680,29 @@ class JavaJarExpansionService(object):
   argument which will spawn a subprocess using this jar to expand the
   transform.
   """
-  def __init__(self, path_to_jar, extra_args=None):
-    if extra_args is None:
-      extra_args = ['{{PORT}}', f'--filesToStage={path_to_jar}']
+  def __init__(self, path_to_jar, extra_args=None, classpath=None):
     self._path_to_jar = path_to_jar
     self._extra_args = extra_args
+    self._classpath = classpath
     self._service_count = 0
 
+  def _default_args(self):
+    to_stage = ','.join([self._path_to_jar] + sum(
```

Comment:
Fail here if self._path_to_jar is not set ?

Reply:
It will always be set. This is a private method, only factored out for testing.